### PR TITLE
Transaction modal

### PR DESCRIPTION
**Pull request by jjjhill**
- Transaction modal for confirming transactions
- Rework contracts to use web3.eth.Contract instead of truffle-contract
- Implement dAppState model and show connect buttons on shop/crypt when not connected
  - Remove OwnerBalances and buttons in shop when wallet is not connected
- Infinite scrolling in shop using @mchigit's pagination code

image
image
image

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -153,13 +161,45 @@ export default {
...
+      const cryptozContractAddress = cryptozArtifact.networks[networkId].address;
+      const czxpContractAddress = czxpArtifact.networks[networkId].address;
+
+      const cryptozContract = new web3.eth.Contract(cryptozArtifact.abi, cryptozContractAddress)
+      const czxpContract = new web3.eth.Contract(czxpArtifact.abi, czxpContractAddress)
+
+      console.log({ cryptozContract, czxpContract })
+      return this.$store.dispatch('setContractInstance', { cryptoz: cryptozContract, czxp: czxpContract })
+    },
+    handleConnect: async function() {
```
Comment (mchigit) [inline src/App.vue:196]: When does `handleConnect` get called? Will we be calling `initializeApp` twice? Once in created and once in handleConnect?

**Comment by ryanprice** [comment]
What do you think about popping up the "pending modal" when the "Connect to blockchain" button is clicked ? is that easy to pop in ?

**Comment by mchigit** [comment]
Some other thoughts:

1. Should we not let user close the modal? If I just close the transaction modal, I'm in a state where I can still use the site but will be in pending transaction state: 
image

2. Sometimes, the isOwned prop is not added because `rootState.web3.coinbase` is null. Results in this:
image
I can mint every single card

3. buy booster credit modal button looks a bit off:
image

4. Sometimes, "already minted" button doesn't show up for me:
image

Everytime I refresh the shop, the amount of card I can mint is different. Is this something only I'm seeing?

File: src/App.vue (hunk 1 above)
Comment (jjjhill) [reply src/App.vue:196]: handleConnect is called by clicking the "connect" buttons

**Comment by jjjhill** [comment]
> Some other thoughts:
> 
> 1. Should we not let user close the modal? If I just close the transaction modal, I'm in a state where I can still use the site but will be in pending transaction state:
I don't think we should remove the X button on the modal, it seems like bad UX. If they close it and start another transaction, metamask will tell them there's 2 pending transactions
> 2. Sometimes, the isOwned prop is not added because `rootState.web3.coinbase` is null. Results in this:
>    I can mint every single card
Cannot reproduce
> 4. Sometimes, "already minted" button doesn't show up for me:
Cannot reproduce

**Comment by jjjhill** [comment]
> What do you think about popping up the "pending modal" when the "Connect to blockchain" button is clicked ? is that easy to pop in ?

I can do it, but the modal says "Please confirm the transaction" and connecting isn't really a transaction.. what should it say instead?

**Review by mchigit — APPROVED** [review]
LGTM

File: src/components/CryptContent.vue [hunk 2]
```diff
@@ -1,96 +1,97 @@
...
-        forever.
-      </p>
-
-      <!-- Loads cards here -->
-      <div class="row">
-        <div class="col">
-          <b-button
-            v-b-tooltip.hover="'Mint 1 random booster NFT'"
-            class="btn btn-danger"
-            v-bind:disabled="boostersOwned < 1"
```
Comment (ryanprice) [inline src/components/CryptContent.vue:73]: is this working ? my button is enabled with zero booster credits

**Review by ryanprice — APPROVED** [review]
Awesome work man, looks great

**Pull request merged**